src/SourceGenerators/Uno.UI.Tasks/LinkerHintsGenerator/LinkerDefinitionMergerTask.cs

```diff
@@ -64,7 +64,31 @@ public override bool Execute()
 					linkerNode.InnerXml += defDoc.DocumentElement.InnerXml;
 				}
 
-				doc.Save(TargetDefinitionFile);
+				string? existingContent;
+				try
+				{
+					existingContent = File.ReadAllText(TargetDefinitionFile);
+				}
+				catch (FileNotFoundException)
+				{
+					// We use catch instead of File.Exists in case it could get deleted in between.
+					existingContent = null;
+				}
+
+				using (var writer = new StringWriter())
+				{
+					doc.Save(writer);
+					var output = writer.ToString();
```

Comment:
How big can this get? I see the benefits of saving only when data has changed, but what about memory consumption, keeping two copies (latest and new) versions of the file in memory?

Reply:
I wouldn't think the cost here is noticeable, especially compared to the benefits we get.

Reply:
Good question Andres, but it's it's not a large file, it can be kept in memory easily.